### PR TITLE
feat: misc enhancements

### PR DESCRIPTION
**Pull request by Elessar1802**
# Description
Fixes for multiple minor FE issue

Fixes https://github.com/devtron-labs/sprint-tasks/issues/1287

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Pages/Applications/DevtronApps/Details/AppConfigurations/MainContent/DeploymentConfigCompare/DeploymentConfigCompare.tsx [hunk 1]
```diff
@@ -480,13 +508,47 @@ export const DeploymentConfigCompare = ({
         if (_isManifestView) {
             setConvertVariables(false)
         }
-        push(
-            generatePath(path, {
+
+        /** NOTE: retain current search params but only append manifestChartRefId(s) if _isManifestView otherwise remove them
+         * This is done to retain selection across jumps between manifest view and config view */
+        const currentSearchParamsObject = Object.fromEntries(new URLSearchParams(location.search).entries())
```
Comment (RohitRaj011) [inline src/Pages/Applications/DevtronApps/Details/AppConfigurations/MainContent/DeploymentConfigCompare/DeploymentConfigCompare.tsx:514]: simplify as discussed

**Review by RohitRaj011 — CHANGES_REQUESTED** [review]

**Review by RohitRaj011 — DISMISSED** [review]

File: src/components/common/navigation/Navigation.tsx [hunk 2]
```diff
@@ -425,7 +425,7 @@ export default class Navigation extends Component<
                                 return this.renderNavLink(item)
                             }
                         })}
-                        {!window._env_.K8S_CLIENT && !this.props.isAirgapped && (
+                        {!window._env_.K8S_CLIENT && !this.props.isAirgapped && this.props.serverMode !== SERVER_MODE.EA_ONLY &&  (
```
Comment (vivek-devtron) [inline src/components/common/navigation/Navigation.tsx:428]: Please add installationType check as well. this should be hidden only in EA-Ent

File: src/services/service.types.ts [hunk 3]
```diff
@@ -139,6 +140,7 @@ export interface AppRouterType {
     isSuperAdmin?: boolean
     appListCount: number
     loginCount: number
+    serverMode: SERVER_MODE
```
Comment (vivek-devtron) [inline src/services/service.types.ts:143]: Please remove this and use from context

File: src/components/common/navigation/NavigationRoutes.tsx [hunk 4]
```diff
@@ -570,17 +580,21 @@ export const AppRouter = ({ isSuperAdmin, appListCount, loginCount }: AppRouterT
...
     const isArgoInstalled: boolean = argoInfoData?.result?.status === ModuleStatus.INSTALLED
 
     return (
         <ErrorBoundary>
             <Switch>
-                <Route path={`${path}/:appType`} render={() => <NewAppList isArgoInstalled={isArgoInstalled} />} />
+                <Route exact path={`${path}/:appType`} render={() => <NewAppList isArgoInstalled={isArgoInstalled} />} />
                 <Route exact path="">
-                    <RedirectToAppList />
+                    {serverMode === SERVER_MODE.EA_ONLY ? (
```
Comment (vivek-devtron) [inline src/components/common/navigation/NavigationRoutes.tsx:593]: Please use the newly created flag

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**